Fix non-interactive options for apt-get and zypper.

zeylos opened the pull request:
##### Summary
<!--
Describe the change in summary section, including rationale and design decisions.
Include "Fixes #nnn" if you are fixing an existing issue.
-->
Modify netdata-updater.sh and kickstart.sh adding force-confdef and force-confold to apt-get so that non-interactive runs and replacing `zypper install -y` by `zypper --non-interactive install`.

##### Test Plan

<!--
Provide enough detail so that your reviewer can understand which test cases you
have covered, and recreate them if necessary. If our CI covers sufficient tests, then state which tests cover the change.
-->

`netdata-updater.sh --non-interactive --not-running-from-cron`
`kickstart.sh --non-interactive`

##### Additional Information

- APT updater error (from discord discussion)
```
[/tmp/netdata-kickstart-2mxUecrtAY]# //usr/libexec/netdata/netdata-updater.sh --non-interactive --not-running-from-cron 
Tue Jun 20 18:31:06 CEST 2023 : INFO: netdata-updater.sh:         [LOG TRUNCATED]        
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
netdata-repo is already the newest version (2-1).
0 upgraded, 0 newly installed, 0 to remove and 116 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up netdata (1.39.0) ...

Configuration file '/etc/netdata/netdata.conf'
 ==> Modified (by you or by a script) since installation.
 ==> Package distributor has shipped an updated version.
   What would you like to do about it ?  Your options are:
    Y or I  : install the package maintainer's version
    N or O  : keep your currently-installed version
      D     : show the differences between the versions
      Z     : start a shell to examine the situation
 The default action is to keep your current version.
*** netdata.conf (Y/I/N/O/D/Z) [default=N] ? 
```
- zypper modifications
  - non-interactive option, from man ↓
  ```
  It’s recommended to use the --non-interactive global option instead. Global options are passed before the command (zypper --non-interactive COMMAND ...). Unlike the no-confirm command option, the global option can be used together with any zypper command.
  ```
  - without `--no-gpg-checks` ↓
 
```
[/tmp/netdata-kickstart-m4PtvKXlcO]# env zypper --non-interactive install --allow-unsigned-rpm /tmp/netdata-kickstart-m4PtvKXlcO/netdata-repo-edge-2-1.noarch.rpm
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following NEW package is going to be installed:
  netdata-repo-edge

1 new package to install.
Overall download size: 6.7 KiB. Already cached: 0 B. After the operation, additional 479.0 B will be used.
Continue? [y/n/v/...? shows all options] (y): y
Retrieving: netdata-repo-edge-2-1.noarch (Plain RPM files cache)                                                                                                           (1/1),   6.7 KiB
netdata-repo-edge-2-1.noarch.rpm:
    Header V4 RSA/SHA512 Signature, key ID f9177b5265f56346: NOKEY

warning: /var/tmp/zypp.hRKBI9/zypper/_tmpRPMcache_/%CLI%/netdata-repo-edge-2-1.noarch.rpm: Header V4 RSA/SHA512 Signature, key ID 65f56346: NOKEY
Looking for gpg key ID 65F56346 in cache /var/cache/zypp/pubkeys.
Repository Plain RPM files cache does not define additional 'gpgkey=' URLs.
netdata-repo-edge-2-1.noarch (Plain RPM files cache): Signature verification failed [4-Signatures public key is not available]
Abort, retry, ignore? [a/r/i] (a): a
Problem occurred during or after installation or removal of packages:
Installation has been aborted as directed.
Please see the above error message for a hint.
 FAILED

 WARNING  Failed to install repository configuration package.

 WARNING  Could not install native binary packages, falling back to alternative installation method.

  ```

<!-- This is usually used to help others understand your
motivation behind this change. A step-by-step reproduction of the problem is
helpful if there is no related issue. -->
  <!--
<details> <summary>For users: How does this change affect me?</summary>

Describe the PR affects users: 
- Which area of Netdata is affected by the change?
- Can they see the change or is it an under the hood? If they can see it, where?
- How is the user impacted by the change? 
- What are there any benefits of the change? 

</details>
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/netdata/netdata?pullRequest=15288) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/netdata/netdata?pullRequest=15288) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/netdata/netdata?pullRequest=15288) it.</sub>

andrewm4894 commented:
@zeylos first PR whohooo!!!

🎈 🎉

Ferroin reviewed (commented):
Overall LGTM and does what it says.

I think we need similar changes in the kickstart script as well though for consistency. The relevant code there starts around line 1380 currently. @zeylos if you can add that to this PR it would be greatly appreciated.

zeylos commented:
Added similar changed to kickstart.sh ↓
- Introduced `install_subcmd` var
- Reworked the quotes on apt-get install_subcmd because of some weird `dpkg: error: need an action option`
- Added `--no-gpg-checks` on zypper install because it seems that `allow-unsigned-rpm` is not enough (repository install aborted every time in non-interactive mode

Fixed stuff on netdata-updater.sh
- renamed `upgrade_subcmd`  to match the other variables names.
- matched quotes on apt-get `install_subcmd`

Tested both on ubuntu, fedora and opensuse containers.

Dim-P commented on packaging/installer/kickstart.sh:
```diff
@@ -1415,6 +1421,11 @@ try_package_install() {
+      if [ "${INTERACTIVE}" = "0" ]; then
+        install_subcmd="--non-interactive --no-gpg-checks install"
+      else
+        install_subcmd="install"
```
Thank you for your contribution @zeylos !

Since you touched this part of the code, adding `--no-gpg-checks` to `install_subcmd="install"` should act as a temporary fix for https://github.com/netdata/netdata/issues/14256 .

Dim-P replied:
I guess since you've already got 2 approvals, we can leave this one for a separate PR.

zeylos replied:
Oh I'm so sorry I didn't read well ! I'm adding it to the interactive part as well

zeylos commented:
Hi guys,

Is anything else blocking the merge ?

zeylos commented:
Sorry guys I dissmissed the reviews, added temporay fix for https://github.com/netdata/netdata/issues/14256 as well, as asked by @Dim-P

thiagoftsm commented:
Thank you very much for your contribution @zeylos !
